Agoric fails to build

Command from our heighliner build server:
```
heighliner build -b --buildkit-addr tcp://127.0.0.1:8125 -r ghcr.io/strangelove-ventures/heighliner -c agoric -v pismoD
```

Error:
```
 => ERROR [linux/amd64->arm64 build-env  9/13] RUN set -eux;    if [ ! -z "1.18.10" ]; then export PATH=/usr/local/cargo/bin:/usr/local/sbin:/usr/local/b  119.2s
 => CACHED [linux/amd64 build-env  5/13] RUN git clone -b pismoD --single-branch https://github.com/Agoric/agoric-sdk.git --recursive                        0.0s
 => CACHED [linux/amd64 build-env  6/13] WORKDIR /build/agoric-sdk                                                                                           0.0s
 => CACHED [linux/amd64 build-env  7/13] RUN if [ ! -z "" ]; then      if [ ! -z "" ]; then cd ""; fi;      if [ ! -f "Cargo.toml" ]; then exit 0; fi;       0.0s
 => CACHED [linux/amd64 build-env  8/13] RUN set -eux;     if [ ! -z "1.18.10" ]; then      wget https://go.dev/dl/go1.18.10.linux-amd64.tar.gz  -O - | tar  0.0s
 => [linux/amd64 build-env  9/13] RUN set -eux;    if [ ! -z "1.18.10" ]; then export PATH=/usr/local/cargo/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:  119.4s
 => => # 7771c926464a98b0a902087fdc1bddabd5
 => => # + go build -v -mod=readonly -tags BUILD_TAGS=muslc -ldflags -X github.com/cosmos/cosmos-sdk/version.Name=agoric   -X github.com/cosmos/cosmos-sdk/versio
 => => # n.AppName=agd   -X github.com/cosmos/cosmos-sdk/version.Version=pismoD   -X github.com/cosmos/cosmos-sdk/version.BuildTags=BUILD_TAGS=muslc   -X github.
 => => # com/cosmos/cosmos-sdk/version.Commit=f760927771c926464a98b0a902087fdc1bddabd5 -buildmode=c-shared -o build/libagcosmosdaemon.so ./cmd/libdaemon/main.go
 => => # go: downloading github.com/agoric-labs/tendermint v0.34.23-alpha.agoric.3
 => => # go: downloading github.com/agoric-labs/cosmos-sdk v0.45.11-alpha.agoric.1.1
error building docker image for agoric from ref: pismoD - failed to solve: process "/bin/sh -c set -eux;    if [ ! -z \"$GO_VERSION\" ]; then export PATH=$PATH:/usr/local/go/bin; fi;    if [ \"$TARGETARCH\" = \"arm64\" ]; then export ARCH=aarch64 CAPS=AARCH64;    elif [ \"$TARGETARCH\" = \"amd64\" ]; then export ARCH=x86_64 CAPS=x86_64; fi;    export CARGO_BUILD_TARGET=${ARCH}-unknown-linux-gnu;    if [ \"$TARGETARCH\" != \"$BUILDARCH\" ]; then      export CARGO_TARGET_${CAPS}_UNKNOWN_LINUX_GNU_LINKER=${ARCH}-linux-gnu-gcc        CC_${ARCH}_unknown_linux_gnu=${ARCH}-linux-gnu-gcc        CXX_${ARCH}_unknown_linux_gnu=${ARCH}-linux-gnu-g++        PKG_CONFIG_SYSROOT_DIR=/usr/${ARCH}-linux-gnu;    fi;    [ ! -z \"$PRE_BUILD\" ] && sh -c \"${PRE_BUILD}\";    if [ ! -z \"$BUILD_TARGET\" ]; then      if [ ! -z \"$BUILD_ENV\" ]; then export ${BUILD_ENV}; fi;      if [ ! -z \"$BUILD_TAGS\" ]; then export \"${BUILD_TAGS}\"; fi;      if [ ! -z \"$BUILD_DIR\" ]; then cd \"${BUILD_DIR}\"; fi;      sh -c \"${BUILD_TARGET}\";    fi" did not complete successfully: exit code: 1

panic: Some images failed to build

goroutine 1 [running]:
github.com/strangelove-ventures/heighliner/builder.(*HeighlinerBuilder).BuildImages(0xc0003acfa0)
        /home/strangelove/heighliner/builder/builder.go:490 +0x118
github.com/strangelove-ventures/heighliner/cmd.queueAndBuild({{0x7ffc26e83678, 0x27}, 0x0, 0x1, {0x7ffc26e83660, 0x14}, {0xf415e8, 0x17}, 0x0, 0x0}, ...)
        /home/strangelove/heighliner/cmd/queue.go:119 +0xdca
github.com/strangelove-ventures/heighliner/cmd.BuildCmd.func1(0xc0001d5b80?, {0xf2dcac?, 0x9?, 0x9?})
        /home/strangelove/heighliner/cmd/build.go:130 +0x298
github.com/spf13/cobra.(*Command).execute(0xc0001d5b80, {0xc0001462d0, 0x9, 0x9})
        /home/strangelove/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:860 +0x663
github.com/spf13/cobra.(*Command).ExecuteC(0xc0001d5900)
        /home/strangelove/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:974 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
        /home/strangelove/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:902
github.com/strangelove-ventures/heighliner/cmd.Execute({0x16fa440?, 0xda41c0?, 0xc000002340?})
        /home/strangelove/heighliner/cmd/root.go:32 +0xe5
main.main()
        /home/strangelove/heighliner/main.go:13 +0x2e
```

--- a/chains.yaml
+++ b/chains.yaml
@@ -40,6 +40,8 @@
       -X "github.com/cosmos/cosmos-sdk/version.BuildTags=$BUILD_TAGS" \
       -X github.com/cosmos/cosmos-sdk/version.Commit=$COMMIT"
 
+    export CGO_ENABLED=1
+
     # Build daemon lib
     go build -v -mod=readonly -tags "$BUILD_TAGS" -ldflags "$COMMON_LDFLAGS" -buildmode=c-shared -o build/libagcosmosdaemon.so ./cmd/libdaemon/main.go
 

--- a/chains.yaml
+++ b/chains.yaml
@@ -17,6 +17,7 @@
   build-env:
     - LEDGER_ENABLED=false
     - BUILD_TAGS=muslc
+  # using pre-build so that cargo fetch does not run
   pre-build: |
     apt update && apt install -y python3 g++
 
@@ -28,19 +29,16 @@
 
     set -eux
     cd golang/cosmos
-    export GOOS=linux GOARCH=${TARGETARCH}
-    npm install node-addon-api --legacy-peer-deps
-    # make -j$(nproc)
 
-     COMMIT=$(git log -1 --format='%H')
+    export CGO_ENABLED=1 GOOS=linux GOARCH=${TARGETARCH} CC=${ARCH}-linux-gnu-gcc CXX=${ARCH}-linux-gnu-g++
+
+    COMMIT=$(git log -1 --format='%H')
 
     COMMON_LDFLAGS="-X github.com/cosmos/cosmos-sdk/version.Name=agoric \
       -X github.com/cosmos/cosmos-sdk/version.AppName=agd \
       -X github.com/cosmos/cosmos-sdk/version.Version=$VERSION \
       -X "github.com/cosmos/cosmos-sdk/version.BuildTags=$BUILD_TAGS" \
       -X github.com/cosmos/cosmos-sdk/version.Commit=$COMMIT"
-
-    export CGO_ENABLED=1
 
     # Build daemon lib
     go build -v -mod=readonly -tags "$BUILD_TAGS" -ldflags "$COMMON_LDFLAGS" -buildmode=c-shared -o build/libagcosmosdaemon.so ./cmd/libdaemon/main.go
@@ -53,6 +51,8 @@
     go build -v -mod=readonly -tags "$BUILD_TAGS" -ldflags "$COMMON_LDFLAGS" -o ../../bin/agd ./cmd/agd
 
     # Build node deps
+    export CC=${ARCH}-linux-gnu-gcc CXX=${ARCH}-linux-gnu-g++ CC_host=gcc CXX_host=g++
+    npm install node-addon-api --legacy-peer-deps
     make node-compile-gyp
     cd ../../packages
     rm -rf ui-components web-components wallet-connection wallet web-components

--- a/chains.yaml
+++ b/chains.yaml
@@ -16,9 +16,7 @@
   github-repo: agoric-sdk
   build-env:
     - LEDGER_ENABLED=false
-    - BUILD_TAGS=muslc
-  # using pre-build so that cargo fetch does not run
-  pre-build: |
+  build-target: |
     apt update && apt install -y python3 g++
 
     curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.3/install.sh | bash
@@ -29,11 +27,8 @@
 
     set -eux
     cd golang/cosmos
-
     export CGO_ENABLED=1 GOOS=linux GOARCH=${TARGETARCH} CC=${ARCH}-linux-gnu-gcc CXX=${ARCH}-linux-gnu-g++
-
     COMMIT=$(git log -1 --format='%H')
-
     COMMON_LDFLAGS="-X github.com/cosmos/cosmos-sdk/version.Name=agoric \
       -X github.com/cosmos/cosmos-sdk/version.AppName=agd \
       -X github.com/cosmos/cosmos-sdk/version.Version=$VERSION \
@@ -49,17 +44,18 @@
 
     # Build agd
     go build -v -mod=readonly -tags "$BUILD_TAGS" -ldflags "$COMMON_LDFLAGS" -o ../../bin/agd ./cmd/agd
-
-    # Build node deps
-    export CC=${ARCH}-linux-gnu-gcc CXX=${ARCH}-linux-gnu-g++ CC_host=gcc CXX_host=g++
+  
+    # Build nodejs swingset kernel
     npm install node-addon-api --legacy-peer-deps
-    make node-compile-gyp
+    export CC_host=gcc CXX_host=g++
+    GYP_DEBUG="--arch=${TARGETARCH}" make node-compile-gyp
+    # cp binding.gyp.in binding.gyp
+    # /root/.nvm/versions/node/*/lib/node_modules/npm/bin/node-gyp-bin/node-gyp configure build --arch=${TARGETARCH}
     cd ../../packages
     rm -rf ui-components web-components wallet-connection wallet web-components
     cd ..
-    npm i -g yarn
-    yarn
-    yarn build
+    npm_config_arch=${TARGETARCH} yarn
+    npm_config_arch=${TARGETARCH} yarn build
 
     # Move to final location
     mkdir /agoric-sdk

